internal/batches/executor/partial_eval.go

```diff
@@ -0,0 +1,388 @@
+package executor
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+	"text/template"
+	"text/template/parse"
+)
+
+// isStaticBool parses the input as a text/template and attempts to evaluate it
+// with only the ahead-of-execution information available in StepContext.
+//
+// To do that it first calls parseAndPartialEval to evaluate the template as
+// much as possible.
+//
+// If, after evaluation, more than text is left (i.e. because the template
+// requires information that's only available later) the function returns with
+// the first return value being false, because the template is not "static".
+// first return value is true.
+//
+// If only text is left we check whether that text equals "true". The result of
+// that check is the second return value.
+func isStaticBool(input string, ctx *StepContext) (bool, bool, error) {
```

Comment:
I'm not a big fan of named return arguments, but this might be a case where they'd be useful for documentation purposes, at least?

Reply:
Good point, changed!